Storage/ThirdParty/SwiftData.swift

```diff
@@ -609,13 +624,27 @@ open class ConcreteSQLiteDBConnection: SQLiteDBConnection {
             return nil
         }
 
-        let status = immediately ? sqlite3_close(db) : sqlite3_close_v2(db)
+        var status = sqlite3_close(db)
```

Comment:
Yes, I spoke with Richard about this yesterday. `_close_v2` always returns "OK" even if the DB couldn't close immediately. By doing a normal `_close` first, we can actually see if anything is preventing the DB from closing, log it, then do `_close_v2`.